src/utils/cache.ts

```diff
@@ -8,21 +8,27 @@ export function cacheIdentifierForPayload(
   const simpleParams: string[] = skipBlockRef
     ? paramsWithoutBlockTag(payload)
     : payload.params ?? [];
-  if (canCache(payload)) {
+  if (canCache(payload.method)) {
     return `${payload.method}:${stringify(simpleParams)}`;
   }
   return null;
 }
 
-export function canCache(payload: Payload): boolean {
-  return cacheTypeForPayload(payload) !== 'never';
+/**
+ * Return whether a method can be cached or not.
+ *
+ * @param method - The method to check.
+ * @returns Whether the method can be cached.
+ */
+export function canCache(method?: string): boolean {
```

Comment:
Why is this argument optional? Same for the other functions here.

If this can be undefined because `Payload.method` itself is optional, then I'm wondering whether this would make more sense conceptually:

``` suggestion
export function canCache(method: string | undefined): boolean {
```

That way `canCache(undefined)` would be a valid way to call this function, but `canCache()` would not.

Reply:
Yeah it's because of the Payload type, as you guessed. I hadn't bothered with this because I'm going to make this mandatory again once we get rid of Payload. I've almost gotten rid of it already in the latest PR.

Reply:
The soonest I can remove this is in #182. I've just updated that PR with a commit to do just that: https://github.com/MetaMask/eth-json-rpc-middleware/pull/182/commits/bbd4931b6f7889a6a72903c0c7c1db4186374390

Reply:
Gotcha, makes sense.